src/test/java/tests/ShoppingCartTest.java

```diff
@@ -0,0 +1,58 @@
+package tests;
+
+import com.epam.tamentoring.bo.Product;
+import com.epam.tamentoring.bo.ShoppingCart;
+import com.epam.tamentoring.exceptions.ProductNotFoundException;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.*;
+
+public class ShoppingCartTest {
+
+    @Test
+    public void testAddProductsToShoppingCart() {
+        ShoppingCart shoppingCart = new ShoppingCart(new ArrayList<>());
+        Product product = new Product();
+        shoppingCart.addProductToCart(product);
+        assertThat(shoppingCart.getProducts()).contains(product);
+    }
+
+    @Test
+    public void testAddProductsToShoppingCartViaConstructor() {
+        List<Product> products = new ArrayList<>();
+        Product product = new Product();
+        products.add(product);
+        ShoppingCart shoppingCart = new ShoppingCart(products);
+        shoppingCart.addProductToCart(product);
```

Comment:
Why do we need it while we are testing adding via constructor?
After this test we would never know if was it added via the constructor or via the method.

Reply:
Yes, indeed, the line 29 was unnecessary. Fixed.